lib/site-inspector.rb

```diff
@@ -127,37 +134,213 @@ def to_json
     to_hash.to_json
   end
 
-  def to_hash
+  def http
+    details = {
+      # site-inspector's best guess
+      canonical: uri.to_s,
+      canonical_protocol: https? ? :https : :http,
+      canonical_endpoint: www? ? :www : :root,
+
+      endpoints: endpoints
+    }
+
+    combos = details[:endpoints]
+
+    details[:live] = (
+      combos[:https][:www][:live] or
+      combos[:https][:root][:live] or
+      combos[:http][:www][:live] or
+      combos[:http][:root][:live]
+    )
+
+    details[:broken_root] = (
+      (combos[:https][:root][:status] == 0) and
+      (combos[:http][:root][:status] == 0)
+    )
+
+    details[:broken_www] = (
+      (combos[:https][:www][:status] == 0) and
+      (combos[:http][:www][:status] == 0)
+    )
+
+    details[:enforce_https] = (
+      (
+        (combos[:http][:www][:status] == 0) ||
+        (combos[:http][:www][:redirect_to_https])
+      ) and
+      (
+        (combos[:http][:root][:status] == 0) or
+        (combos[:http][:root][:redirect_to_https])
+      ) and
+      (
+        combos[:https][:www][:live] or
+        combos[:https][:root][:live]
+      )
+    )
+
+    details[:redirect] = (
+      combos[:http][:www][:redirect_to_external] and
+      combos[:http][:root][:redirect_to_external] and
+      combos[:https][:www][:redirect_to_external] and
+      combos[:https][:root][:redirect_to_external]
+    )
+
+    # HSTS on the canonical domain? (valid HTTPS checked in endpoint)
+    details[:hsts] = combos[:https][details[:canonical_endpoint]][:hsts]
+
+    # HSTS on the entire domain?
+    details[:hsts_entire_domain] = (
+      combos[:https][:root][:hsts] and
+      combos[:https][:root][:hsts_header].downcase.include?("includesubdomains")
+    )
+
+    # HSTS preload --ready?
+    details[:hsts_entire_domain_preload] = (
+      details[:hsts_entire_domain] and
+      combos[:https][:root][:hsts_header].downcase.include?("preload")
+    )
```

Comment:
Whether a domain has its entire domain ready for preloading requires that it include all subdomains, and that it include the `preload` directive (this is not specified in the RFC, but [Google's HSTS preload form](https://hstspreload.appspot.com/) -- the RFC allows for custom directives).

It is possible to preload an individual domain or subdomain without specifying `includeSubDomains` (and `healthcare.gov` currently does this), but this only measures when an entire domain is "done".